### PR TITLE
Feature/hm 67 edit property by id

### PR DESCRIPTION
**Pull request by viktorgkw**

### CODE REVIEW

File: src/Properties/Properties.Api/Controllers/PropertiesController.cs [hunk 1]
```diff
@@ -135,5 +136,49 @@ public async Task<IActionResult> Delete(int id)
                     return BadRequest();
             }
         }
+
+        [HttpPut]
+        [Route("{id}")]
+        [Authorize(Roles = UserRoles.Seller + "," + UserRoles.Broker)]
```
Comment (rosteslav) [inline src/Properties/Properties.Api/Controllers/PropertiesController.cs]: $"{...},{...}" - interpolation, always better

File: src/Properties/Properties.Api/Controllers/PropertiesController.cs [hunk 2]
```diff
@@ -135,5 +136,49 @@ public async Task<IActionResult> Delete(int id)
...
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> Edit(
+            int id,
```
Comment (rosteslav) [inline src/Properties/Properties.Api/Controllers/PropertiesController.cs]: From What?

File: src/Properties/Properties.Infrastructure/Migrations/20231125173815_InitialCreate.cs
Comment (rosteslav) [inline src/Properties/Properties.Infrastructure/Migrations/20231125173815_InitialCreate.cs:1]: Don't touch auto-gen stuff!

File: src/Properties/Properties.Infrastructure/Repositories/PropertiesRepository.cs [hunk 3]
```diff
@@ -144,6 +144,29 @@ await _context.Properties
             await _context.SaveChangesAsync();
         }
 
+        public async Task EditById(int id, AddPropertyInputModel editedProperty)
+        {
+            var entityToUpdate = await _context.Properties
+                .FirstAsync(e => e.Id == id);
+
+            entityToUpdate.NumberOfRooms = editedProperty.NumberOfRooms;
```
Comment (rosteslav) [inline src/Properties/Properties.Infrastructure/Repositories/PropertiesRepository.cs]: We don't do that here! Mapping so not done this way!

File: src/Properties/Properties.Application/Mapper/PropertyProfile.cs [hunk 4]
```diff
@@ -16,7 +16,7 @@ public PropertyProfile()
             CreateMap<Property, PropertyModel>()
                 .ForMember(x => x.CreatedOnLocalTime, opt => opt.MapFrom(src => src.CreatedOnUtcTime.ToLocalTime()))
                 .IncludeAllDerived();
-
```
Comment (rosteslav) [inline src/Properties/Properties.Application/Mapper/PropertyProfile.cs:19]: what only this non-adding-value change and not doing any mapping here?

**Review by rosteslav — CHANGES_REQUESTED** [review]

File: src/Properties/Properties.Infrastructure/Migrations/20231125173815_InitialCreate.cs
Comment (viktorgkw) [reply src/Properties/Properties.Infrastructure/Migrations/20231125173815_InitialCreate.cs:1]: Visual Studio Cleanup did

Comment (rosteslav) [reply src/Properties/Properties.Infrastructure/Migrations/20231125173815_InitialCreate.cs:1]: Your changes, your responsibility! Before submitting look at your changes! Do you want all of them? Something forgotten?

File: src/Properties/Properties.Infrastructure/Repositories/PropertiesRepository.cs [hunk 5]
```diff
@@ -144,6 +144,16 @@ await _context.Properties
             await _context.SaveChangesAsync();
         }
 
+        public async Task EditById(int id, AddPropertyInputModel editedProperty)
+        {   
+            var entityToUpdate = await _context.Properties
+                .FirstAsync(e => e.Id == id);
+
+            _mapper.Map(editedProperty, entityToUpdate);
```
Comment (rosteslav) [inline src/Properties/Properties.Infrastructure/Repositories/PropertiesRepository.cs]: I like this way, readability
var result = _mapper.Map<IEnumerable<ImagesResult>>(images);

Comment (rosteslav) [reply src/Properties/Properties.Infrastructure/Repositories/PropertiesRepository.cs]: do you have that mapper?

**Review by rosteslav — CHANGES_REQUESTED** [review]

File: src/Properties/Properties.Infrastructure/Repositories/PropertiesRepository.cs (hunk 5 above)
Comment (viktorgkw) [reply src/Properties/Properties.Infrastructure/Repositories/PropertiesRepository.cs]: Yes, the repository accepts it from the primary constructor and we store it in a private readonly field.

**Review by rosteslav — APPROVED** [review]

**Pull request merged**